src/event-cache/__tests__/EventCache.integ.test.ts

```diff
@@ -75,7 +75,7 @@ describe('EventCache tests', () => {
         });
     });
 
-    test('meta data contains default attributes not overridden from custom attributes', async () => {
+    test('default meta data attributes except platformType can be overriden by custom attributes', async () => {
```

Comment:
question: Why make an exception for platform type?

Reply:
It was the main one that I couldn't think of a valid use case to overwrite, but if it makes sense to allow it I can update this

Reply:
(2) seems like the closest thing to a two-way-door decision, as I feel like it wouldn't be too painful to revert it with a major version bump.

Do you have important use cases for overwriting other attributes, which would require (3)?